ocaml/sdk-gen/csharp/autogen/src/JsonRpc.cs

```diff
@@ -155,6 +158,11 @@ public partial class JsonRpcClient
     {
         private int _globalId;
 
+#if (NET462_OR_GREATER || NETSTANDARD2_0_OR_GREATER)
+        // TODO: get proper version string from build
```

Comment:
We never merge source with TODOs on `master`.

Use the name of the package `XenServer.NET` and its version. I think we can get it from https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assemblyname.version?view=netstandard-2.0, but needs testing

If not, this needs a placeholder, and changes to pipelines. We can advise on that if needed

Reply:
If the client code uses the C# SDK as a project in their code, `typeof(JsonRpcClient).Assembly.GetName().Version` should return the SDK version, but in the corner case of someone dropping the sources in a client project (like XenCenter does), it should return the client project version (in XC it would return XenModel's version, which is XC's version).

If we can't have the ocaml build use a placeholder and replace it with the tag (we ran into some trouble with @psafont when we tried it for the API reference), I suggest using `new ActivitySource("XenAPI.JsonRpcClient", "@SDK_VERSION@")` (the truth is I dislike placeholders like this, but we have one more insance of it in the PS code).

By the way, is it indeed the SDK version we need here? (As opposed to a string that will be representing the implementing client, somehting like the UserAgent, for example).

Reply:
> but in the corner case of someone dropping the sources in a client project (like XenCenter does)


we could do without the placeholder and just add a line in the README :shrug:

Reply:
> By the way, is it indeed the SDK version we need here? (As opposed to a string that will be representing the implementing client, somehting like the UserAgent, for example).

for this, I'd say so

if a client wants to use open telemetry, it should add it itself and then use its own user agent

I may be wrong - I'm not that familiar with what the expected behaviour is

Reply:
> it should add it itself and then use its own user agent

Yes, the thing is we offer the option to use own UserAgent (albeit it sub-optimally designed in my opinion), but we don't offer this option for the activity source with this design.